### ISSUE TITLE
Calendar A11y Pass

### ISSUE THREAD

**Issue by awalia13** [OfficeDev/office-ui-fabric-react#8349]
Tracking all of the issues brought up with the 03/15 accessibility pass on Calendar fabric control
Details on individual bugs can be found on [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B8cfe67c5-ce76-4083-b60e-4190dbd38206%7D&action=default&uid=%7B8CFE67C5-CE76-4083-B60E-4190DBD38206%7D&ListItemId=1872&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)
- [x] ~Bug1: [Accessibility][Programmatic access]: Name is not defined for arrow mark which is used for changing the month.~ Fixed in #9195.
- [x] ~Bug2: [Accessibility][Keyboard Navigation]: Focus moves to top of the page whenever user press Enter on 'Go to Today' Button.~ Fixed.
- [ ] Bug3: [Accessibility][Usable]: In Scan mode screen reader focus is going to the calendar dates twice. Tracked in #11140
- [ ] Bug4: [Accessibility][Screen Readers]: Two months in the month Picker calendar is having aria-selected property as true when only one is selected by user. Tracked in #11141
- [ ] Bug5: [Accessibility][Screen Readers]: Information about the change of week is not narrated by screen reader whenever user press Enter on previous/next button. Tracked in #11142
- [x] ~Bug6: [Accessibility][Keyboard Navigation]: Overlaid Year Picker is not accessible by Keyboard.~ Fixed.
- [ ] Bug7: [Accessibility][Keyboard Navigation]: Upon Selecting the date, calendar dialog box blinks and focus moves to the arrow. Tracked in #11143
- [x] ~Bug8: [Accessibility][Keyboard Navigation]: Focus is moving out of the calendar dialog if Go to Today button is disabled.~ Fixed.
- [ ] Bug9: [Accessibility][Programmatic access]: Name is not defined for the dialog which appears on pressing enter on click for calendar button. Tracked in #11144
- [ ] ~Bug10: [Accessibility][Usable][Suggestion]: Role=Application is used for calendars. #11145~
- [x] ~Bug11: [Accessibility][Usable]: Screen does not scroll down along with keyboard focus in Backward navigation.~ Fixed. 
- [ ] Bug12: [Accessibility][Usable]: Calendar dialog does not close with ESC in scan mode of narrator. Tracked in #9713

**Comment by staciecarroll** [comment]
(1) should be fixed by PR in #9195

**Comment by dollykumari5** [comment]
Bug 1, Bug 2, Bug 6, Bug 8 and Bug 11 from the above list of bugs are fixed. 
Verified on Test Environment:    
OS: Version: 1903(OS Build 18362.100)   
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.10.0/#/examples/calendar   
Browser: Edge

**Comment by JllyGrnGiant** [comment]
Is there any information about what the fix for Bug10 (role=application) would entail? It isn't clear to me what is supposed to be done instead besides removing the role, which, from what I can tell (I could be mistaken), would make screen readers switch to scan/browse mode when they reach the table.

**Comment by lorejoh12** [comment]
> Is there any information about what the fix for Bug10 (role=application) would entail? It isn't clear to me what is supposed to be done besides removing the role, which, from what I can tell (I could be mistaken), would make screen readers switch to scan/browse mode when they reach the table.

From my understanding, role=application means the controls do not conform to normal WAI-ARIA widgets, so it manages it's own focus, keyboard, input, etc. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Application_Role

This is the case here, the control does handle all of that itself, which makes it seem like the role is appropriate? It sounds like just having a region with role=application somewhere on the page is breaking something else for them in Sharepoint, but without more details I'm not sure what role they would prefer.

**Comment by 999-Dev** [comment]
New Issue Found :
Bug13: [Usable :Calendar Panel]: Narrator reads incorrect info on dates like "Read only on dates".

Test Environment: 
OS: Version: RS5_Prerelease(1903 OS Build 18990.1) 
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/calendar
Edge Browser Version: 
Microsoft Edge: 44.18990.1.0
Microsoft EdgeHTML: 18.18990
Chrome Browser Version:
Chrome Version: 76.0.3809.132

**Comment by dzearing** [comment]
All existing bugs are now tracked individually. Closing.
